### DISCUSSION TITLE
Picklist dependent value

### DISCUSSION THREAD

**Discussion by drakub** [YetiForceCompany/YetiForceCRM#15982] [General] [answered]
I have a field [Category] with many picklist values. I have a priority field with values [Low, Medium, High, Urgent]. I would like to save the priority [Low] when the category selected is [cat1, cat3, cat5, etc] or save [Medium] when category is [cat2, cat6, cat10, etc] or [High] when category is [cat4, cat8, cat9, etc].
I tried using a workflow, the challenge is that the workflow will make me set several conditions since my category list over 2000. 
I tried using picklist dependency but not a desired approach because I don't want the Priority to be a selectable field - it should be done in the backend.

How can I achieve this behavior

**Comment by vovpff** [comment]
You can write handler what will change picklist for your needs. This needs the programmer

**Comment by drakub** [comment]
I have written a handler already but I am getting challenges when I tried
to fetch the picklist values from database; Any clues;

*Draku Job*
*Tel: +256 786 610580*
*B. Information Systems*
*Systems Developer | Database Designer |**Technical Writer | ICT Instructor*

**Reply by vovpff** [reply]
There a lot of ways how to implement it. I think you need 3 arrays for priority filled with categories. Then category changed you need to search for new category value into arrays. If category found then priority should be changed to some value

**Comment by drakub** [comment]
Thanks a lot.

I have worked this out using a handler

*Draku Job*
*Tel: +256 786 610580*
*B. Information Systems*
*Systems Developer | Database Designer |**Technical Writer | ICT Instructor*

**Comment by drakub** [comment, marked as answer]
I am managed to resolve this issue by implementing a custom handler
